semgrep/semgrep/core_runner.py

```diff
@@ -88,61 +79,35 @@ def _raise_semgrep_error_from_json(
         elif error_type == "invalid regexp in rule":
             raise SemgrepError(f'Invalid regexp in rule: {error_json["message"]}')
         elif error_type == "invalid pattern":
-            if self._optimizations == "all":
-                range = error_json["range"]
-                s = error_json.get("pattern", "<no pattern>")
-                matching_span = Span.from_string_token(
-                    s=s,
-                    line=range.get("line", 0),
-                    col=range.get("col", 0),
-                    path=range.get("path", []),
-                    filename="semgrep temp file",
-                )
-                if error_json["message"] == "Parsing.Parse_error":
-                    long_msg = f"Pattern `{s.strip()}` could not be parsed as a {error_json['language']} semgrep pattern"
-                else:
-                    long_msg = f"Error parsing {error_json['language']} pattern: {error_json['message']}"
-
-                raise InvalidPatternError(
-                    short_msg=error_type,
-                    long_msg=long_msg,
-                    spans=[matching_span],
-                    help=None,
-                )
-            # no special formatting ought to be required for the other types; the semgrep python should be performing
+            range = error_json["range"]
+            s = error_json.get("pattern", "<no pattern>")
+            matching_span = Span.from_string_token(
+                s=s,
+                line=range.get("line", 0),
+                col=range.get("col", 0),
+                path=range.get("path", []),
+                filename="semgrep temp file",
+            )
+            if error_json["message"] == "Parsing.Parse_error":
+                long_msg = f"Pattern `{s.strip()}` could not be parsed as a {error_json['language']} semgrep pattern"
             else:
-                matching_pattern = next(
-                    (p for p in patterns if p._id == error_json["pattern_id"]), None
```

Comment:
Per pattern errors are not something that can happen now that we do not run per-pattern